### ISSUE TITLE
Meaning/usability of CGAL_CXX_FLAGS_INIT

### ISSUE THREAD

**Issue by joachim-reichel** [CGAL/cgal#2734]
Original bug report: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=886670

When CGAL is installed on a Debian system from .deb packages, CGAL_CXX_FLAGS_INIT in CGALConfig.cmake is set to "-g -O2 -fdebug-prefix-map=/build/cgal-d6DBFP/cgal-4.11=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -frounding-math".

These flags seem to be the CXXFLAGS that were use to build the CGAL library itself. They are the result of distribution-specific defaults for the package build environment, only the last flag was added by CGAL itself.These values are also used when building the demos/examples and when building real applications, e.g., via "find_package( CGAL QUIET COMPONENTS  ); include( ${CGAL_USE_FILE} )".

However, almost all of the flags in CGAL_CXX_FLAGS_INIT are not relevant for real applications using CGAL. Things like "-g -O2" should not be decided by a library. Exposing "-fdebug-prefix-map=..." is meaningless after the CGAL library was built. Various hardening flags (and warning flags in general) should also not be decided by a library. The only flag that is CGAL-specific (and important for CGAL users) is -frounding-math.

I believe it would be nicer for CGAL users if the cmake snippets would just contain the flags that are necessary to use CGAL, and keep these flags separate from the flags that were used to build the library (why are they kept/exposed anyway?).

Not sure what's best for the demos/examples in this context: treat them like applications (only -frounding-math) or use the flags used to build the library itself.

* Operating system: Debian stretch
* Compiler: g++ 6.x
* CGAL version: 4.11

**Comment by mglisse** [comment]
I think Laurent's message in December: "Heads up!: Important changes in the CMake scripts of CGAL" says this is fixed in master, but you may want to check.

**Comment by lrineau** [comment]
That is right. I have started to modernize our CMake scripts, to slowly converge to a regular usage of CMake, instead of our hacks and tweaks that dated from the time of CMake-2.4.

That patch is only in `master`, and that means CGAL-4.12. I am reluctant to backport a rework of our CMake scripts to a release branch like `releases/CGAL-4.11-branch`, because that would mean a lot of work and manual testing.

For the records, that was fixed by #1436.
